spk/firebird/src/service-setup.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+FBGUARD_BIN_FILE="${SYNOPKG_PKGDEST}/bin/fbguard"
+
+# We set onetime because we want the systemd to take care about the failures
+SERVICE_COMMAND="${FBGUARD_BIN_FILE} -pidfile ${PID_FILE} -daemon -onetime"
+
+SYSDBA_PASSWORD_FILE="${SYNOPKG_PKGDEST}/SYSDBA.password"
+
+service_postinst () {
+    set -x
+    cd "$SYNOPKG_PKGDEST" || exit 1
+    bin/gbak -rep msg.gbak msg.fdb
+    bin/build_file
+
+    # Make the message file accessible so other users can use isql
+    chmod go+r firebird.msg
+
+    bin/gbak -rep security4.gbak security4.fdb
+
+    echo "Remove unnecessary intermediate files"
+    rm -f bin/build_file msg.gbak msg.fdb security4.gbak
+
+    if [ ! -f "${SYSDBA_PASSWORD_FILE}" ];then
+        SYSDBA_PASSWORD="${wizard_sysdba_password}"
+
+        echo "Setting 'SYSDBA' password to '${SYSDBA_PASSWORD}'";
+
+        echo "create or alter user SYSDBA password '$SYSDBA_PASSWORD' using plugin Srp; commit; quit;" \
+           | bin/isql -user sysdba security.db
```

Comment:
IMHO also security.db file must be located in the folder ${SYNOPKG_PKGVAR}.

Probably there must be a check in service_postinst function whether it is called for "INSTALL" or "UPGRADE" to execute different code.

Reply:
Doing so requires knowing the `${SYNOPKG_PKGVAR}` value in `cross/firebird/Makefile` as the security db location must be compiled in the sources. It loos like the variable is set first in the `spk/firebird/Makefile` which is too late.
What is the best way to work around this? Would including `../../common.service.mk` in the cross/... Makefile be enough?

Reply:
In the (cross) Makefile you have the variables `$(INSTALL_PREFIX_VAR)` and `$(STAGING_INSTALL_PREFIX_VAR)` available. 
Those can be used to define `CONFIGURE_ARGS`, but you must adapt it for your very customized build process...

Reply:
Done, the security database is now preserved across upgrades.